### ISSUE TITLE
build-issues: hidapi on windows

### ISSUE THREAD

**Issue by k9ert** [cryptoadvance/specter-desktop#830]
On the build-agent after migrating to python >3.8, we currently have this behaviour:

```
>pip3 install hidapi==0.7.99.post21       
Collecting hidapi==0.7.99.post21
  Using cached hidapi-0.7.99.post21.tar.gz (51 kB)
Requirement already satisfied: setuptools>=19.0 in c:\users\...\test\specter-desktop\.env\lib\site-packages (from hidapi
==0.7.99.post21) (51.0.0)
Building wheels for collected packages: hidapi
  Building wheel for hidapi (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\...\test\specter-desktop\.env\Scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.arg
v[0] = '"'"'C:\\Users\\...\\AppData\\Local\\Temp\\pip-install-e42k_r28\\hidapi_df0dff070c08470dbba33b89659bdc65\\setup.p
y'"'"'; __file__='"'"'C:\\Users\\...\\AppData\\Local\\Temp\\pip-install-e42k_r28\\hidapi_df0dff070c08470dbba33b89659bdc6
5\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.cl
ose();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\...\AppData\Local\Temp\pip-wheel-yt3prdtm'
       cwd: C:\Users\...\AppData\Local\Temp\pip-install-e42k_r28\hidapi_df0dff070c08470dbba33b89659bdc65\
  Complete output (8 lines):
  running bdist_wheel
  running build
  running build_ext
  cythoning hid.pyx to hid.c
  c:\users\...\appdata\local\temp\pip-install-e42k_r28\hidapi_df0dff070c08470dbba33b89659bdc65\.eggs\cython-0.29.21-py3.
8-win-amd64.egg\Cython\Compiler\Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2).
 This will change in a later release! File: C:\Users\...\AppData\Local\Temp\pip-install-e42k_r28\hidapi_df0dff070c08470d
bba33b89659bdc65\hid.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  building 'hid' extension
  error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.micr
osoft.com/visual-cpp-build-tools/
  ----------------------------------------
  ERROR: Failed building wheel for hidapi
  Running setup.py clean for hidapi
Failed to build hidapi
Installing collected packages: hidapi
    Running setup.py install for hidapi ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\...\test\specter-desktop\.env\Scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.a
rgv[0] = '"'"'C:\\Users\\...\\AppData\\Local\\Temp\\pip-install-e42k_r28\\hidapi_df0dff070c08470dbba33b89659bdc65\\setup
.py'"'"'; __file__='"'"'C:\\Users\\...\\AppData\\Local\\Temp\\pip-install-e42k_r28\\hidapi_df0dff070c08470dbba33b89659bd
c65\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.
close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\...\AppData\Local\Temp\pip-record-bp7j
bqg8\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\...\test\specter-desk
top\.env\include\site\python3.8\hidapi'
         cwd: C:\Users\...\AppData\Local\Temp\pip-install-e42k_r28\hidapi_df0dff070c08470dbba33b89659bdc65\
    Complete output (6 lines):
    running install
    running build
    running build_ext
    skipping 'hid.c' Cython extension (up-to-date)
    building 'hid' extension
    error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.mi
crosoft.com/visual-cpp-build-tools/
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\...\test\specter-desktop\.env\Scripts\python.exe' -u -c 'import
 sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\...\\AppData\\Local\\Temp\\pip-install-e42k_r28\\hidapi_df0dff
070c08470dbba33b89659bdc65\\setup.py'"'"'; __file__='"'"'C:\\Users\\...\\AppData\\Local\\Temp\\pip-install-e42k_r28\\hid
api_df0dff070c08470dbba33b89659bdc65\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace
('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\...\A
ppData\Local\Temp\pip-record-bp7jbqg8\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:
\Users\...\test\specter-desktop\.env\include\site\python3.8\hidapi' Check the logs for full command output.
WARNING: You are using pip version 20.3.1; however, version 20.3.3 is available.
You should consider upgrading via the 'C:\Users\...\test\specter-desktop\.env\Scripts\python.exe -m pip install --upgrad
e pip' command.
```

We're currently installing this rather old version of hidapi because of HWI (1.2.0). This is also the case in the newest version of HWI (1.2.1):
https://github.com/bitcoin-core/HWI/blob/1.2.1/setup.py#L122
(btw, HWI does have [this python-requires](https://github.com/bitcoin-core/HWI/blob/1.2.1/setup.py#L150): `'python_requires': '>=3.6,<3.9',`)
```
'hidapi>=0.7.99,<0.8.0',
```
So it seems we need to make that version working. The question is ... how did it worked before? 
The build for v0.10.3 ran through [successfully](https://gitlab.com/cryptoadvance/specter-desktop/-/jobs/893327459):
```
$ python -V
Python 3.7.9
...
Collecting hidapi==0.7.99.post21
  Downloading hidapi-0.7.99.post21-cp37-cp37m-win_amd64.whl (41 kB)
...
```

According to [this](https://github.com/bitcoin-core/HWI/issues/303) issue, hidapi is dependent on cython which is, from our point of view, a pip-package to embed a C-Compiler and so it's itself dependent on a C-Compiler, on Windows typically MinGW. The Installation-instructions for MinGW (commandline-version) are [here](http://www.mingw.org/wiki/Getting_Started).

**Comment by k9ert** [comment]
So after learning a bit about [wheel packages](https://realpython.com/python-wheels/), it seems that the problem is that:
* You either need [VisualStudio](https://realpython.com/python-wheels/#building-a-platform-wheel-macos-and-windows) to build from hidapi-source-package on windows or
* The old version 0.7.99.post21 need to provide a wheel-package for windows out of the box or
* HWI is upgrading hidapi

So the latest point needs a bit more investigation:
* So in this [commit](https://github.com/bitcoin-core/HWI/commit/eeadbd6a680601cd7ca3d1e5519012fc36049aef) on Sep 1, 2020  the version-requirements already got upgraded on master
* The 1.2.1 release got released from the 1.x-branch, though
Therefore it's a question when the upgrade will be done on HWI

@achow101 can you maybe help with this?

**Comment by achow101** [comment]
I'm pretty sure it's because it is using the cached source distribution of hidapi. The prebuilt wheels should already have that all compiled, and there are wheels for windows. There shouldn't be any need to be compiling hidapi on windows.

**Comment by k9ert** [comment]
Great, the cython-hidapi-project released a 3.8-wheel for us.
The build is still not working but at least this issue is fixed. Closing!
